Refactor docs [WIP] (#43)
r2u-io/documentation: docs/javascript/augmented-reality.md

---
title: Augmented Reality
---

After adding the script tag to your website, the methods for creating the Augmented Reality experience will be available through the global `R2U` object.

<p float="left">
  <img src="https://storage.googleapis.com/r2u-sdk-bucket/documentation/product-ar.gif" title="AR" width="200"/>
</p>

## R2U.ar.attach

:::tip `mobile`
:::

The `ar.attach` method attaches an event listener to open the AR experience on mobile devices (e.g.: on a button click). It will automatically track button clicks for analytics purposes.

```typescript
// test SKU -- remember to use your product information
const arButton = document.getElementById('ar-button')
const sku = 'RE000001'
// creates an unsupported device alert
const fallbackOptions = {
  //add message in the alert
  alertMessage: 'AR not supported by device',
  //opens a 3D viewer on the warning screen
  fallback: 'viewer'
}

> R2U.ar.attach({
>   element: arButton,
>   sku,
>   fallbackOptions
>   /* resize defaults to `false` */
> })

Reviewer: Pensando no Troubleshooting da Nai

```suggestion
R2U.ar.attach({
  element: arButton,
  sku: sku,
  fallbackOptions: fallbackOptions
  /* resize defaults to `false` */
})
```

Author: nossa, não acredito que teve gente que errou isso hahaha
vou manter

Reviewer: Na real eles não erram quando é o mesmo nome, mas tem gente que faz:
```ts
R2U.ar.attach({
  arButton,
  sku,
  fallbackOptions
})
```  
Por causa disso, e não existe `arButton` quando era pra ser `element`. Acho que o `sku` e `fallbackOptions` pode ser meio overkill mesmo..

#### showInstructions

:::info Default `once`
:::

When rendered, shows a full-screen panel explaining how to place and manipulate a 3D object, before proceeding to the AR experience.

| value  | effect                                                                            |
| ------ | --------------------------------------------------------------------------------- |
| once   | The instructions screen will render only on the user's first AR experience.       |
| always | The instructions screen will render every time the user interacts with the button |
| never  | The instructions screen will never render                                         |

<div >
<div >

```typescript
interface R2U {
  ar: {
    attach: (params: {
      element: HTMLElement
      sku: string
      showInstructions?: 'once' | 'always' | 'never'
    }) => Promise<void>
  }
}
```

</div>
<div>
<strong> Instructions example: </strong>

<p float="left">
  <img src="https://storage.googleapis.com/r2u-sdk-bucket/documentation/instruction-en.png" title="Instruction screen" width="200"/>
</p>
</div>
</div>

### fallbackOptions

:::tip The `alertMessage` message can be customized
:::

> Shows an image when the mobile device does not support the AR experience. If the instructions parameter is provided, the fallback is displayed only when you click on the "See at your place" button.

Reviewer: ```suggestion
Shows an image when the mobile device does not support the AR experience. If the instructions parameter is provided, the fallback is displayed only when you click on the attached button.
```
